compiler/env/OMRAOTAdapter.hpp

```diff
@@ -0,0 +1,223 @@
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef OMR_AOTADAPTER_INCL
+#define OMR_AOTADAPTER_INCL
+#include "env/TRMemory.hpp"
+#include "env/TypedAllocator.hpp"
+#include "infra/Annotations.hpp"
+#include "compiler/env/jittypes.h"
+#include <map>
+#ifndef OMR_AOTADAPTER_CONNECTOR
+#define OMR_AOTADAPTER_CONNECTOR
+namespace OMR { class AOTAdapter; }
+namespace OMR { typedef OMR::AOTAdapter AOTAdapterConnector; }
+#endif
+
+namespace TR 
+   {
+   class AOTAdapter;
+   class AOTRelocationRuntime;
+   class AOTStorageInterface;
+   class CodeCache;
+   class CodeCacheManager;
+   class RelocationRuntime;
+   class CompilerEnv;
+   class AOTMethodHeader;
+   }
+
+namespace OMR
+{
+
+/**
+ * @brief OMRAotAdapter, an extensible class that is designed to dispatch 
+ * the interaction between compiler components to perform persisting of 
+ * relocatable code. The benefit of having this class is in decoupling AOT-related
+ * classes and removing indirect responsibility by introducing new separation of
+ * responsibilities: the RelocationRuntime class is only concerned with managing 
+ * the generation and application of relocations, without knowing about persisting
+ * methods implemented in the current runtime environment. Similarly, Shared Cache
+ * or other options for persisting code could implement a simplistic interface
+ * (that, obviously can be extended) that would have two functions: receive a message
+ * that contains a serialized version of the AOTMethod header, as well as send one.
+ * The AOTAdapter will register the header and make it available to RelocationRuntime
+ * and hence, the compiler, as well as to runtime environment, since AOTAdapter is
+ * planned to be alive longer than a compilation thread.
+ */
+
+class OMR_EXTENSIBLE AOTAdapter
+   {
+public:
+   TR_ALLOC(TR_Memory::AOTAdapter)
+   AOTAdapter();
+   ~AOTAdapter();
```

Comment:
Thank you for the comment. That's  a copying artifact from my attempt to change map inside of OMRAOTAdapter being keyed by const char* instead of std::string. Removed!